cmd/resolve-version/main_test.go

```diff
@@ -122,3 +124,58 @@ func TestMatchReleaseSemver(t *testing.T) {
 	assert.False(t, result.matched)
 	assert.Equal(t, result.versionRequirement, "99.x")
 }
+
+func genYarnS3ObjectList(versions []string) []s3Object {
+	out := []s3Object{}
```

Comment:
Suggestion for the future:
```
var out []s3Object
// or, since you know the length of versions already
out := make([]S3Object,0,len(versions))
```

append handes nil slices just fine. len(<a nil slice>) is 0, etc.
The make version of this pre-allocates a capacity == len(versions), so the append won't need to allocate a new backing array every so often.

For an example, take a look at : https://play.golang.org/p/y9juBtWlr71

Reply:
I meant to apply this comment to `resolveYarn` above, but it applies to this as well.